### PR TITLE
Add logging for contact information updates

### PR DESCRIPTION
**Pull request by Tonksthebear**
## Summary
Per this [thread](https://dsva.slack.com/archives/C03DY3FN7HQ/p1680878445173539), we need to be logging certain information when there are updates to certain user information

## Related issue(s)
[department-of-veterans-affairs/va.gov-team#5333](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/5333)

## What areas of the site does it impact?
None

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

Is this the correct logging we want

### CODE REVIEW

**Comment by joeniquette** [comment]
@Tonksthebear can you send me an example of the fields that will be published as part of this log? You should be able to run it locally and generate a stub log with the attributes.

**Comment by Tonksthebear** [comment]
@joeniquette sorry for the delay, this led me down quite the rabbit hole. It seems there is a different in how we log things based on if the authentication method is IAM vs SIS. I'm navigating how to get the data depending on the authentication type. I'll post back when I have a solution

**Comment by Tonksthebear** [comment]
@joeniquette just got the differences all sorted out. This is what the logging would look like:

```
"Mobile::V0::EmailsController#create request completed"
{:user_uuid=>"b6c9c0dd-10f7-47ac-98c4-6c36154d90bd", :sso_cookie_contents=>{"patientIcn"=>"1013062086V794840", "signIn"=>{"serviceName"=>"idme", "clientId"=>"101213eb98a2408f85822b3b03a51084", "authBroker"=>"sis"}, "credential_used"=>"idme", "expirationTime"=>"2023-04-18T18:03:51Z"}, :request_host=>"www.example.com"}
```

**Comment by joeniquette** [comment]
This looks like it should work. I'll know for sure once I get the logs in datadog. I'll test once this is merged.

File: modules/mobile/app/controllers/mobile/v0/profile_base_controller.rb [hunk 1]
```diff
@@ -22,6 +24,53 @@ def render_transaction_to_json(transaction)
...
+        Rails.logger.warn(
+          "#{action} request completed", sso_logging_info
+        )
+      end
+
+      def sso_logging_info
+        { user_uuid: @current_user&.uuid,
+          sso_cookie_contents: sso_cookie_content,
+          request_host: request.host }
+      end
```
Comment (aherzberg) [inline modules/mobile/app/controllers/mobile/v0/profile_base_controller.rb:42]: could this sso_logging info be included from elsewhere in vets-api or did you have to do some customization to get it to work in mobile?

**Review by aherzberg — APPROVED** [review]
LGTM

File: modules/mobile/app/controllers/mobile/v0/profile_base_controller.rb (hunk 1 above)
Comment (Tonksthebear) [reply modules/mobile/app/controllers/mobile/v0/profile_base_controller.rb:42]: It needs to be in a controller and the module from vets-api that already contains it overwrites things we use in mobile, so we can't just include the module

**Comment by jperk51** [comment]
Is there a reason this couldn't be done in the services mention in the ticket?

**Comment by Tonksthebear** [comment]
> Is there a reason this couldn't be done in the services mention in the ticket?

Ya, the logging required information most easily found in the controller (probably why the other team implemented it there as well). We could change the services to allow passing those values in (`@current_user`, `@session_object`), but that seemed ugly and like it would affect more dependencies. I didn't want to mess with any initializations of the services, and the other option would be manually setting those values after initialization. Neither seems great

**Review by ryan-mcneil — APPROVED** [review]
Looks good!

**Pull request merged**